### PR TITLE
[MINOR]: Simplify enforce_distribution, minor changes

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

## Rationale for this change
This PR simplifies codebase in enforce_distribution and other places a bit to improve readability. Also couples of unnecessary clones are removed.

### CODE REVIEW

**Review by ozankabak — APPROVED** [review]
LGTM, cleanups FTW

File: datafusion/core/src/physical_optimizer/enforce_sorting.rs [hunk 1]
```diff
@@ -496,9 +496,10 @@ fn ensure_sorting(
     {
         // This SortPreservingMergeExec is unnecessary, input already has a
         // single partition.
+        sort_onwards.truncate(1);
         return Ok(Transformed::Yes(PlanWithCorrespondingSort {
-            plan: children[0].clone(),
-            sort_onwards: vec![sort_onwards[0].clone()],
+            plan: children.swap_remove(0),
```
Comment (alamb) [inline datafusion/core/src/physical_optimizer/enforce_sorting.rs:501]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap_remove says "does not preserve the order" -- is the idea here that there is only one element in child and this avoids a clone?

I think we have used the `pop` method to do something similar elsewhere in the code (like `children.pop().unwrap()`)

Not that this needs to change, just an observation

**Review by alamb — APPROVED** [review]
Thank you @mustafasrepo  and @ozankabak -- looks like a nice cleanup to me

File: datafusion/core/src/physical_optimizer/enforce_sorting.rs (hunk 1 above)
Comment (ozankabak) [reply datafusion/core/src/physical_optimizer/enforce_sorting.rs:501]: Yes exactly, your observation is correct -- in this case `children.pop().unwrap()` and `swap_remove(0)` is equivalent.

**Pull request merged**